src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/EnumerableExtensions.cs

```diff
@@ -105,6 +105,17 @@ public static async Task ForEachAsync<T>(this IEnumerable<T> list, Func<T, Task>
             }
         }
 
+        public static IEnumerable<T> WhereAsync<T>(this IEnumerable<T> list, Func<T, Task<bool>> filter)
+        {
+            var tasks = Task.WhenAll<(T item, bool isSuccess)>(
```

Comment:
Cool 👍 , but need to remember we can't use this with an IQueryable, only an in-memory collection. Since DbContext isn't thread safe, EF core won't run multiple requests in parallel and errors instead. 

I think that's why you've had to add `.ToList` on `filesExcludingReplacements `?

Reply:
Yep pretty much. I'll add a comment to make sure people realise this, although I think running this method would most likely cause LINQ to blow up anyway.

Reply:
Cool, a comment sounds good. Yes it would, you get an error saying that a second operation can't run until an existing operation has completed, or something like that!

Reply:
Added a docblock for the method now